lib/applications.js

```diff
@@ -1,15 +1,27 @@
 'use strict';
 
+const semver = require('semver');
+
+const common = require('./common');
 const errors = require('./errors');
 
 // Generic application class. You are free to substitute it with whatever suits
 // your needs.
-//   name - application name
-//   api - application API
+//   name - application name that may contain version after '@'
+//          (e.g. app@1.0.0). Version in name is preferred over
+//          'version' parameter
+//   api - object that contains interfaces as its fields each of which
+//         contains functions by method names. Each method has the following
+//         signature (connection, <0 or more method arguments>, callback)
+//   version - application version of 'name' application
 //
 class Application {
-  constructor(name, api) {
-    this.name = name;
+  constructor(name, api, version) {
```

Comment:
Even though I agree that it will look better first of all it will not be compatible and second in case version is specified in the `name` you will have to pass `null` to the version parameter, because api will come after it.